src/Workspaces/Core/Portable/Workspace/Solution/Checksum_Factory.cs

```diff
@@ -14,13 +14,50 @@ namespace Microsoft.CodeAnalysis
     // all these are just helper methods
     internal partial class Checksum
     {
+        private static readonly Stack<IncrementalHash> s_hashStack = new Stack<IncrementalHash>();
```

Comment:
oh. sorry, I think you are talking about we free but dropping case where we can't just drop for this hash provider. it require to be disposed when dropped.